cumulus/parachains/runtimes/bridge-hubs/bridge-hub-westend/src/xcm_config.rs

```diff
@@ -0,0 +1,326 @@
+pub type TrustedTeleporters = ConcreteAssetFromSystem<WestendLocation>;
+
+pub struct XcmConfig;
+impl xcm_executor::Config for XcmConfig {
+	type RuntimeCall = RuntimeCall;
+	type XcmSender = XcmRouter;
+	type AssetTransactor = CurrencyTransactor;
+	type OriginConverter = XcmOriginToTransactDispatchOrigin;
+	// BridgeHub does not recognize a reserve location for any asset. Users must teleport Native
+	// token where allowed (e.g. with the Relay Chain).
+	type IsReserve = ();
+	type IsTeleporter = TrustedTeleporters;
+	type UniversalLocation = UniversalLocation;
+	type Barrier = Barrier;
+	type Weigher = WeightInfoBounds<
+		crate::weights::xcm::BridgeHubWestendXcmWeight<RuntimeCall>,
+		RuntimeCall,
+		MaxInstructions,
+	>;
+	type Trader =
+		UsingComponents<WeightToFee, WestendLocation, AccountId, Balances, ToStakingPot<Runtime>>;
+	type ResponseHandler = PolkadotXcm;
+	type AssetTrap = PolkadotXcm;
+	type AssetLocker = ();
+	type AssetExchanger = ();
+	type AssetClaims = PolkadotXcm;
+	type SubscriptionService = PolkadotXcm;
+	type PalletInstancesInfo = AllPalletsWithSystem;
+	type MaxAssetsIntoHolding = MaxAssetsIntoHolding;
+	type FeeManager = XcmFeesToAccount<Self, WaivedLocations, AccountId, TreasuryAccount>;
+	type MessageExporter = (crate::bridge_to_rococo_config::ToBridgeHubRococoHaulBlobExporter,);
+	type UniversalAliases = Nothing;
+	type CallDispatcher = WithOriginFilter<SafeCallFilter>;
+	type SafeCallFilter = SafeCallFilter;
+	type Aliasers = Nothing;
+}
+
+pub type PriceForParentDelivery =
+	ExponentialPrice<FeeAssetId, BaseDeliveryFee, TransactionByteFee, ParachainSystem>;
+
+/// Converts a local signed origin into an XCM multilocation.
+/// Forms the basis for local origins sending/executing XCMs.
+pub type LocalOriginToLocation = SignedToAccountId32<RuntimeOrigin, AccountId, RelayNetwork>;
+
+/// The means for routing XCM messages which are not for local execution into the right message
+/// queues.
+pub type XcmRouter = WithUniqueTopic<(
+	// Two routers - use UMP to communicate with the relay chain:
+	cumulus_primitives_utility::ParentAsUmp<ParachainSystem, PolkadotXcm, PriceForParentDelivery>,
+	// ..and XCMP to communicate with the sibling chains.
+	XcmpQueue,
+)>;
+
+#[cfg(feature = "runtime-benchmarks")]
+parameter_types! {
+	pub ReachableDest: Option<MultiLocation> = Some(Parent.into());
+}
+
+impl pallet_xcm::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type XcmRouter = XcmRouter;
+	// We want to disallow users sending (arbitrary) XCMs from this chain.
+	type SendXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, ()>;
+	// We support local origins dispatching XCM executions in principle...
+	type ExecuteXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, LocalOriginToLocation>;
+	type XcmExecuteFilter = Nothing;
+	type XcmExecutor = XcmExecutor<XcmConfig>;
+	type XcmTeleportFilter = Everything;
+	type XcmReserveTransferFilter = Nothing; // This parachain is not meant as a reserve location.
```

Comment:
But the AH `type IsReserve` says that it recognizes the Bridge Hub for ROC/WND. Maybe leave a comment that the reserve location is actually the foreign chain, but we trust the Bridge Hub to relay the message that a reserve is being held.

Reply:
`XcmReserveTransferFilter` is used by `pallet-xcm` to filter which assets are allowed to be reserve transferred using pallet-xcm local extrinsic call.

`Nothing` is right here - I would even suggest we change
```diff
- type XcmTeleportFilter = Everything;
+ // only WND is allowed for teleportation using pallet-xcm
+ type XcmTeleportFilter = NativeAsset;
```

> the AH type `IsReserve` says that it recognizes the Bridge Hub for ROC/WND

actually `IsReserve` on AH says that it recognizes the other side's AH as trusted reserve. BHs should not be trusted reserves for anything.

Reply:
ok, changed docs for AH

Reply:
> ```diff
> + // only WND is allowed for teleportation using pallet-xcm
> + type XcmTeleportFilter = NativeAsset;
> ```

there is just AssetTransactor for native token, but I think we can be also explicit here, but `NativeAsset` does not implement `Contains<(MultiLocation, Vec<MultiAsset>)>` which is `XcmTeleportFilter`, but with `LocationWithAssetFilters` it is doable, I will mark TODO